### PR TITLE
[refactor] RepositoryLocation -> CodeLocation

### PR DESCRIPTION
**Pull request by smackesey**
## Summary & Motivation

Internal companion PR: https://github.com/dagster-io/internal/pull/5220

Rename all variants of "repository location" to "code location" equivalent. This is good because:

- repository is already quasi-deprecated and will eventually be removed
- we refer to these things as "code locations" publicly and in our docs

I left the GQL API alone since I'm still fuzzy on what can be changed here and what can't be.

I have included a large list of renames in this one PR (full list below). They are broken into individual commits. Ideally we can keep this in one PR because many of these renames have internal equivalents, so breaking this up would require a large stack of parallel OSS/internal PRs.

In general each commit is just a rename using Pylance rename symbol, though there was sometimes followup manual work to convert untyped references (which pylance can't identify).

Most renames targeted symbols not whitelisted for serdes. For the few that were, I used `storage_name`/`storage_field_names` `whitelist_for_serdes` args to ensure backcompatibility:

- `<new name> (<storage name>)`
- `InProcessCodeLocationOrigin ("InProcessRepositoryLocationOrigin")`
- `RegisteredCodeLocationOrigin ("RegisteredRepositoryLocationOrigin")`
- `ManagedGrpcPythonEnvCodeLocationOrigin ("ManagedGrpcPythonEnvRepositoryLocationOrigin")`
- `GrpcServerCodeLocationOrigin ("GrpcServerRepositoryLocationOrigin")`
- `ExternalRepositoryOrigin.code_location_origin ("repository_location_origin")`
- `NotebookPathArgs.code_location_origin ("repository_location_origin")`

Full list of renames:

- `DagsterRepositoryLocationNotFoundError` -> `DagsterCodeLocationNotFoundError`
- `DagsterRepositoryLocationLoadError` -> `DagsterCodeLocationLoadError`
- BaseWorkspaceRequestContext.get_external_notebook_data(..., repository_location_name, ...) -> code_location_name
- dagster_k8s_test_infra.integration_utils.launch_run_over_graphql(..., repository_location_name, ...) -> code_location_name
- [local-variables] repository_location(s) -> code_location(s)
- _api.snapshot_repository(..., repository_location, ...) -> code_location
- WorkspaceCodeLocationEntry.repository_location -> code_location
- _utils.external.external_pipeline_from_location(..., repo_location, ...) -> code_location
- _core.execution.job_backfill.create_backfill_run(..., repo_location, ...) -> code_location
- _core.execution.job_backfill.submit_backfill_runs(..., repo_location, ...) -> code_location
- DagsterInstance.created_reexecuted_run(..., repo_location, ...) -> code_location
- NotebookPathArgs.repository_location_origin -> code_location_origin
- GrpcServerRegistry <method>(..., repository_location_origin, ...) -> code_location_origin
- _cli.workspace.cli_target.get_repository_location_from_kwargs -> get_code_location_from_kwargs
- WorkspaceProcessContext.repository_locations_count -> code_locations_count
- WorkspaceProcessContext.shutdown_repository_location -> shutdown_code_location
- WorkspaceProcessContext.reload_repository_location -> reload_code_location
- WorkspaceProcessContext.has_repository_location_error -> has_code_location_error
- WorkspaceProcessContext.has_repository_location -> has_code_location
- WorkspaceProcessContext.repository_location_names -> code_location_names
- BaseWorkspaceRequestContext.shutdown_repository_location -> shutdown_code_location
- BaseWorkspaceRequestContext.reload_repository_location -> reload_code_location
- BaseWorkspaceRequestContext.has_repository_location -> has_code_location
- BaseWorkspaceRequestContext.has_repository_location_name -> has_code_location_name
- BaseWorkspaceRequestContext.get_repository_location_error -> get_code_location_error
- BaseWorkspaceRequestContext.has_repository_location_error -> has_code_location_error
- BaseWorkspaceRequestContext.repository_location_errors -> code_location_errors
- BaseWorkspaceRequestContext.repository_locations -> code_locations
- BaseWorkspaceRequestContext.repository_location_names -> code_location_names
- _core.host_representation.repository_location -> code_location
- RepositoryHandle.repository_location -> code_location
- RepositoryHandle.repository_location_origin -> code_location_origin
- workspace.cli_target.get_external_repository_from_repo_location -> get_external_repository_from_code_location
- workspace.cli_target.get_repository_location_from_workspace -> get_code_location_from_workspace
- ExternalRepositoryOrigin.repository_location_origin -> code_location_origin
- IWorkspace.get_location_statuses -> get_code_location_statuses
- IWorkspace.get_repository_location -> get_code_location
- GrpcServerRepositoryLocation -> GrpcServerCodeLocation
- InProcessRepositoryLocation -> InProcessCodeLocation
- GrpcServerRepositoryLocationOrigin -> GrpcServerCodeLocationOrigin
- ManagedGrpcPythonEnvRepositoryLocationOrigin -> ManagedGrpcPythonEnvCodeLocationOrigin
- InProcessRepositoryLocationOrigin -> InProcessCodeLocationOrigin
- RegisteredRepositoryLocationOrigin -> RegisteredCodeLocationOrigin
- WorkspaceLocationStatusEntry -> CodeLocationStatusEntry
- WorkspaceLocationEntry -> CodeLocationEntry
- WorkspaceLocationLoadStatus -> CodeLocationLoadStatus
- RepositoryLocation -> CodeLocation
- RepositoryLocationOrigin -> CodeLocationOrigin

## How I Tested These Changes

Existing test suite.

### CODE REVIEW

**Comment by smackesey** [comment]
Current dependencies on/for this PR:
* master
  * **PR #13024** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13024" target="_blank">Graphite</a> 
    * **PR #13025** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13025" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/13025?utm_source=stack-comment).

File: python_modules/dagster/dagster/_cli/workspace/cli_target.py [hunk 1]
```diff
@@ -496,8 +496,7 @@ def repository_click_options() -> Sequence[ClickOption]:
             "--location",
             "-l",
             help=(
-                "RepositoryLocation within the workspace, necessary if more than one location is"
-                " present."
+                "CodeLocation within the workspace, necessary if more than one location is present."
```
Comment (gibsondan) [inline python_modules/dagster/dagster/_cli/workspace/cli_target.py:499]: Ditch 'workspace' too

Name of the code location, necessary if more than one location is present.

File: python_modules/dagster/dagster/_core/host_representation/origin.py [hunk 2]
```diff
@@ -106,37 +106,39 @@ def location_name(self) -> str:
...
+    def create_location(self) -> "CodeLocation":
         pass
 
 
-@whitelist_for_serdes
-class RegisteredRepositoryLocationOrigin(
-    NamedTuple("RegisteredRepositoryLocationOrigin", [("location_name", str)]),
-    RepositoryLocationOrigin,
+# Different storage name for backcompat
+@whitelist_for_serdes(storage_name="RegisteredRepositoryLocationOrigin")
```
Comment (gibsondan) [inline python_modules/dagster/dagster/_core/host_representation/origin.py:114]: nice

File: python_modules/dagster/dagster/_core/workspace/context.py [hunk 3]
```diff
@@ -600,57 +594,57 @@ def _load_location(self, origin: RepositoryLocationOrigin) -> WorkspaceLocationE
...
             )
 
-        return WorkspaceLocationEntry(
+        return WorkspaceCodeLocationEntry(
             origin=origin,
-            repository_location=location,
+            code_location=location,
             load_error=error,
-            load_status=WorkspaceLocationLoadStatus.LOADED,
+            load_status=WorkspaceCodeLocationLoadStatus.LOADED,
```
Comment (gibsondan) [inline python_modules/dagster/dagster/_core/workspace/context.py:601]: While we are doing sweeping naming changes we should take out "Workspace" too IMO - anywhere that says "WorkspaceCodeLocation" should just be "CodeLocation"

File: python_modules/dagster/dagster/_core/workspace/workspace.py [hunk 4]
```diff
@@ -5,55 +5,55 @@
...
-class WorkspaceLocationLoadStatus(Enum):
+class WorkspaceCodeLocationLoadStatus(Enum):
     LOADING = "LOADING"  # Waiting for location to load or update
     LOADED = "LOADED"  # Finished loading (may be an error)
 
 
-class WorkspaceLocationEntry(NamedTuple):
-    origin: "RepositoryLocationOrigin"
-    repository_location: Optional["RepositoryLocation"]
+class WorkspaceCodeLocationEntry(NamedTuple):
```
Comment (gibsondan) [inline python_modules/dagster/dagster/_core/workspace/workspace.py:17]: CodeLocationEntry

**Review by gibsondan — APPROVED** [review]

File: python_modules/dagster/dagster/_core/workspace/context.py (hunk 3 above)
Comment (smackesey) [reply python_modules/dagster/dagster/_core/workspace/context.py:601]: Sounds good-- any other older conventions you can think of that should be changed? I don't have full context on the meaning/history of all these names.

**Pull request merged**